client/src/use/AttributeTypes.ts

```diff
@@ -0,0 +1,101 @@
+import { LineChartData } from './useLineChart';
+
+export interface NumericAttributeEditorOptions {
+  type: 'combo'| 'slider';
+  range?: number[];
+  steps?: number;
+}
+export interface StringAttributeEditorOptions {
+  type: 'locked'| 'freeform';
+}
+
+export interface AttributeRendering {
+  typeFilter: string[];
+  selected?: boolean;
+  displayName: string;
+  displayColor: 'auto' | string;
+  displayTextSize: number;
+  valueColor: 'auto' | string;
+  valueTextSize: number;
+  order: number;
+  layout: 'vertical' | 'horizontal';
+  location: 'inside' | 'outside';
+  corner?: 'NW' | 'SE' |'SW';
+  box: boolean;
+  boxColor: 'auto' | string;
+  boxThickness: number;
+  boxBackground?: string;
+  boxOpacity?: number;
+  displayWidth: {
+    type: 'px' | '%';
+    val: number;
+  };
+  displayHeight: {
+    type: 'px' | 'auto' | '%';
+    val: number;
+  };
+}
+export interface Attribute {
+  belongs: 'track' | 'detection';
+  datatype: 'text' | 'number' | 'boolean';
+  values?: string[];
+  valueColors?: Record<string, string>;
```

Comment:
`valueColors` is technically a `Record<string | number, string>` the key is either a key provided an attribute value which could be a string, or a numerical key used for the gradients.  The issue is that the d3 typescript believes that `range()` only takes in numerical array.  Their own documentation for the method says it can take in any values supported by the underlying interpolator.  So the strings representing colors works for creating a gradient.
![image](https://github.com/Kitware/dive/assets/61746913/937009cf-f21a-49ae-8b7e-abb37dd3f47f)
Updating the generics for ScaleLinear I think fixes this: `d3.scaleLinear<string, number>()`
I'll update the base type of the valueColors to be Record<string | number, string> and will update this code to use the proper generics.